trust-store-trust-policy-specification.md

```diff
@@ -181,21 +181,110 @@ The implementation must allow the user to execute custom validations. These cust
+        1. Perform extended validation using the applicable(if any) plugin.
+        1. If you have reached this step then treat the OCI artifact signature as a valid signature.
+
+### Certificate Revocation Evaluation
+
+If the certificate revocation trust-store setting is set to `skip`, skip the below steps. Otherwise, check for revocation status for certificate and certificate chain.
+
+1. If the revocation status of any of the certificates cannot be determined (revocation unavailable) and `signingIdentityRevocation` is set to either `enforceWithFailOpen` or `warn` then log a warning and skip the below steps. Otherwise, fail the signature validation and exit.
+1. If any of the certificates are revoked and `signingIdentityRevocation` is set to either `enforceWithFailOpen` or `enforceWithFailClose` then fail signature validation and exit else log a warning.
+
+Starting from Root to leaf certificate, for each certificate in the certificate chain, perform the following steps to check its revocation status:
+
+- If the certificate being validated doesn't include information OCSP or CRLs then no revocation check is performed and the certificate is considered valid(not revoked).
+- If the certificate being validated includes either OCSP or CRL information, then the one which is present is used for revocation check.
+- If both OCSP URLs and CDP URLs are present, then OCSP is preferred over CRLs. If revocation status cannot be determined using OCSP because of any reason such as unavailability then fallback to using CRLs for revocation check.
+
+#### CRLs
+
+There are two types of CRLs (per RFC 3280), Base CRLs and Delta CRls.
+
+- **Base CRLs:** Contains the revocation status of all certificates that have been issued by a given CA. BaseCRLs are signed by the certificate issuing CAs.
+- **Delta CRLs:** Contains only certificates that have changed status since the last base CRL was published. Delta CRLs are signed by the certificate issuing CAs.
+- **Indirect CRLs:** Special Case in which BaseCRLs and Delta CRLs are not signed by the certificate issuing CAs, instead they are signed with a completely different certificate.
+
+Notary v2 MUST support BaseCRLs and Delta CRLs. Notary v2 MAY support Indirect CRLs. Notary v2 supports only HTTP CRL URLs.
+
+Implementations MAY add support for caching CRLs and OCSP response to improve availability, latency and avoid network overhead.
+
+##### CRL Download
+
+CRL download location (URL) can be obtained from the certificate's CRL Distribution Point (CDP) extension. If the certificate contains multiple CDP locations then each location download is attempted in sequential order. For each CDP location, Notary V2 will try to download the CRL for the default threshold of 10 seconds. If the CRL cannot be downloaded within the timeout threshold the revocation result will be "revocation unavailable". The user may be able to configure this threshold.
```

Comment:
Do we want to leave (`The user may be able to configure this threshold.`) open ended for now? Just seems to lead to what should be a link for how this is done today, or a new issue to track for how the user can configure it.
Or, we can simply remove that last sentence and do a subsequent PR when and how configuration is enabled.

Reply:
Yes, keeping it open-ended is intentional. We want the implementations to have some sane defaults and we also want to allow implementations to override these values with user-provided values.

Reply:
Ok, so the notary spec says the user may configure it. 
The current `notation` cli doesn't yet implement it, but the spec says it may.